src/include/function/cast/functions/cast_functions.h

```diff
@@ -60,6 +60,39 @@ inline void CastDateToTimestamp::operation(common::date_t& input, common::timest
     result.value /= common::Interval::MICROS_PER_SEC;
 }
 
+struct CastToDate {
+    template<typename T>
+    static inline void operation(T& input, common::date_t& result);
+};
+
+template<>
+inline void CastToDate::operation(common::date_t& input, common::date_t& result) {
+    result = input;
```

Comment:
Just a quick comment, why do we perform data to date casting? Should this be a noop(no operation)?

Reply:
When casting between nested types, a cast is performed on the children regardless of whether or not they're the same.
```
CREATE NODE TABLE tab(id SERIAL, col1 STRUCT(a DATE, b INT32), PRIMARY KEY(id));
CREATE (t:tab {col1: {a: date('2024-04-05'), b: 12}});
```
An alternative solution to this is checking whether or not children are identical in nested casting.

Reply:
I tend to check check instead adding a X to X cast if checking is not tricky.